jwst/refpix/irs2_subtract_reference.py

```diff
@@ -387,6 +386,86 @@ def decode_mask(output, mask):
     return bits
 
 
+def rm_intermittent_badpix(data, scipix_n, refpix_r):
+    """Find pixels that are intermittently bad and replace with nearest
+    good value.
+
+    Parameters
+    ----------
+    data : 4-D ndarray
+        The data array in detector orientation.  This includes both the
+        science and interleaved reference pixel values.  `data` will be
+        modified in-place. The science data values will not be modified.
+
+    scipix_n : int
+        Number of regular (science) samples before stepping out to collect
+        reference samples.
+
+    refpix_r : int
+        Number of reference samples before stepping back in to collect
+        regular samples.
+
+    Returns
+    -------
+    data : 4-D ndarray
+        The data array in detector orientation.  This includes both the
+        science and interleaved reference pixel values.  The intermittently
+        bad pixels are now set to the nearest good reference pixel value.
+    """
+    # The intermittently bad pixels will be replaced for all integrations
+    # and all groups. The last group will be used to identify them
+    nits, ngroups, ny, nx = np.shape(data)
+    total_rp2replace = []
+    # calculate differences of even and odd pairs per amplifier
+    amplifier = nx // 5                 # 640
+    for k in range(1, 5):
+        diffs, rp2check = [], []
+        offset = int(k * amplifier)
+        # jump from the start of the reference pixel sequence to the next
+        ref_pix, rp2check, pair = [], [], 0
+        for rpstart in range(scipix_n//2, amplifier, scipix_n+refpix_r):
+            rpstart += offset
+            # go through the 4 reference pixels
+            for ri in range(4):
+                ri = rpstart + ri
+                ref_pix.append(ri)
+                rp_val = np.mean(data[nits-1, ngroups-1, :, ri])
+                rp_std = np.std(data[nits-1, ngroups-1, :, ri])
+                if ri % 2 != 0:
+                    odd_pix = ri
+                    rp_odd = rp_val
+                else:
+                    even_pix = ri
+                    rp_even = rp_val
+                pair += 1
+                if pair == 2:
+                    # only do difference if the pixel has not already been flagged as bad
+                    if rp_odd != 0. and rp_even != 0.:
+                        diffs.append(np.abs(rp_odd - rp_even))
+                        rp2check.append(even_pix)
+                        rp2check.append(odd_pix)
+                    pair = 0
+        diff_m = np.mean(np.abs(diffs))
+
+        # order indeces increasing from left to right
```

Comment:
minor spelling issue: should change "indeces" to either "indices" or "indexes"

Reply:
done